modules/tinymce/src/core/main/ts/fmt/RemoveFormat.ts

```diff
@@ -212,10 +207,29 @@ const removeFormat = function (ed: Editor, format: RemoveFormat, vars?: FormatVa
   // "matchName" will made sure we're dealing with an element, so cast as one
   const elm = node as DomElement;
 
+  // Applies to styling elements like strong, em, i, u, etc. so that if they have styling attributes, the attributes can be kept but the styling element is removed
+  if (format.inline && Type.isArray(format.preserve_attributes)) {
```

Comment:
Should this also have a `format.remove === 'all'` check? Or do we want to run this logic when doing `remove: 'empty'` and `remove: 'none'`?

Reply:
Thinking about this more, we definitely don't want this to run on `remove: 'none'` as it won't be removing the element.

Reply:
Yeah, you are right. It makes most sense for it to only work with `format.remove === 'all'`. Will add that check in.